packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -30,20 +30,49 @@ export interface SystemWorkspaceDatastore {
 	};
 }
 
-/**
- * There is no implementation class for this interface.
- * It is a simple structure. Most complicated aspect is that
- * `connectionId()` member is replaced with a new
- * function when a more recent connection is added.
- *
- * See {@link SystemWorkspaceImpl.ensureAttendee}.
- */
-interface SessionClient extends ISessionClient {
+class SessionClient implements ISessionClient {
 	/**
 	 * Order is used to track the most recent client connection
 	 * during a session.
 	 */
-	order: number;
+	public order: number = 0;
+
+	private connectionStatus: SessionClientStatus;
+
+	public constructor(
+		public readonly sessionId: ClientSessionId,
+		private connectionId: ClientConnectionId | undefined = undefined,
+	) {
+		this.connectionStatus =
+			connectionId === undefined
+				? SessionClientStatus.Disconnected
+				: SessionClientStatus.Connected;
+	}
+
+	public getConnectionId(): ClientConnectionId {
+		if (this.connectionId === undefined) {
+			throw new Error("Client has never been connected");
+		}
+		return this.connectionId;
+	}
+
+	public getConnectionStatus(): SessionClientStatus {
+		return this.connectionStatus;
+	}
+
+	public setConnectionId(
+		connectionId: ClientConnectionId,
+		updateStatus: boolean = true,
```

Comment:
When might you set updateStatus to false?

Reply:
We think the management of connectivity is funky at the moment. I expect this will change when [Task 22616](https://dev.azure.com/fluidframework/internal/_workitems/edit/22616): "Address ISessionClient connectivity consistency" is done. I hope this parameter just goes away. We'll see. I am using it now to preserve the logic as-is.